components/core/tests/test-ffi_Value.cpp

```diff
@@ -0,0 +1,195 @@
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <type_traits>
+#include <variant>
+#include <vector>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+
+#include "../src/clp/ffi/encoding_methods.hpp"
+#include "../src/clp/ffi/Value.hpp"
+#include "../src/clp/ir/EncodedTextAst.hpp"
+#include "../src/clp/ir/types.hpp"
+
+using clp::ffi::Value;
+using clp::ffi::value_bool_t;
+using clp::ffi::value_float_t;
+using clp::ffi::value_int_t;
+using clp::ir::eight_byte_encoded_variable_t;
+using clp::ir::EightByteEncodedTextAst;
+using clp::ir::four_byte_encoded_variable_t;
+using clp::ir::FourByteEncodedTextAst;
+using std::string;
+using std::vector;
+
+namespace {
+/**
+ * Parses and encodes the given string as an instance of `EncodedTextAst`.
+ * @tparam encoded_variable_t
+ * @param text
+ * @return The encoded result.
+ */
+template <typename encoded_variable_t>
+requires(std::is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>
+         || std::is_same_v<encoded_variable_t, four_byte_encoded_variable_t>)
+[[nodiscard]] auto get_encoded_text_ast(std::string_view text
+) -> clp::ir::EncodedTextAst<encoded_variable_t>;
+
+/**
+ * Tests querying the underlying type of the given value against the given type.
```

Comment:
```suggestion
 * Tests that `Value::is` returns true for the given type and false for all others.
```

The current docstring is not wrong but I just feel like being more direct is clearer.